processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -251,6 +245,112 @@ public static boolean convertSegment(File toConvert, File converted, IndexSpec i
     public MMappedIndex mapDir(File inDir) throws IOException;
   }
 
+  public static void validateRowValues(
+      Rowboat rb1,
+      IndexableAdapter adapter1,
+      Rowboat rb2,
+      IndexableAdapter adapter2
+  )
+  {
```

Comment:
@himanshug we could add that as well, but the main reason comparing values was necessary when indices didn't match up is because of https://github.com/druid-io/druid/issues/1309.